src/app/shared/components/plh-main-header.ts

```diff
@@ -38,8 +38,16 @@ export class PLHMainHeaderComponent implements OnInit, OnDestroy {
   hardwareBackButton$: PluginListenerHandle;
   /** track if navigation has been used to handle back button click behaviour */
   hasBackHistory = false;
-  constructor(private router: Router, private location: Location) {}
+  constructor(
+    private router: Router,
+    private location: Location,
+    private appConfigService: AppConfigService
+  ) {}
   async ngOnInit() {
+    this.appConfigService.appConfig$.subscribe((appConfig: IAppConfig) => {
+      this.appConfig = appConfig;
```

Comment:
nit(blocking)
I don't quite understand why we need to keep the full appConfig instead of just the header_defaults, so probably a `headerConfig` variable would make more sense that takes the value of the appConfig header_defaults property.
Also as we now are dealing with a display component it would be good not to leave subscriptions open as they can lead to memory leaks, So either you will want to add an unsubscribe method/pipe into ngOnDestroy, or ignore the subscription altogether and just bind the display component to a public `appConfigService.appConfig.APP_HEADER_DEFAULTS.title | async` (as the async pipe automatically handles unsubscribe).

Reply:
The main thing to know about subscriptions is that they use memory when subscribed, so if you have a component that gets destroyed and recreated frequently and subscribes each time on init you quickly lead to memory leaks. This isn't really important for things like global services which hold single subscriptions (or arguably our header component which should mostly persist throughout), but good practice to consider when using display components.

The angular `async` pipe includes custom logic to automatically unsubscribe when a component is destroyed so usually the safest way to go about subscriptions, but definitely not the only way - most common is hooking into the ngOnDestroy method to manually unsubscribe. In order to do this all subscriptions need to be stored as a named variable for reference, and individually called with _unsubscribe()_ on destroy which can become a bit tedious if you have multiple, so a common simplification is to have a single _componentDestroyed$_ emitter (or similar) which can be piped into subscriptions as a means to unsubscribe via `takeUntil` - you can see examples of this in our codebase if you search `destroyed$`

As for variable naming - yeah this gets ugly pretty quickly. The easiest workaround is when using `ngIf` statements you can [rename the condition variable](https://angular.io/api/common/NgIf#storing-a-conditional-result-in-a-variable) (e.g. `ngIf="appConfigService.appConfig | async as config"` and refer to `config["APP_HEADER_DEFAULTS].title` or similar renaming more nested to headerConfig. Otherwise you can use ng-templates to pass variables between components with different names (a bit messy), or as a final quick solution just write a funtion in the `.ts` file to reference the variable, e.g.
```ts
get headerConfig(){
  return this.appConfigService.appConfig$.value.APP_HEADER_DEFAULTS`
}
```
and simply reference as `<div>{{headerConfig.title}}</div>`

Reply:
I can see you've gone ahead with adding the unsubscribe method which makes sense so feel free to mark as resolved